### ISSUE TITLE
Pass ecmaVersion to detective (To enable support for async/await +)

### ISSUE THREAD

**Issue by brandonhorst** [browserify/module-deps#125]
Node v7 supports async/await with `--harmony-async-await`. Chrome stable supports it now. module-deps (and therefore Browserify) chokes on it.

Could `module-deps` accept an input parameter to set the `ecmaVersion` property on the call to `detective()`? If it was set to 8, browserify could handle async/await just fine.

**Comment by goto-bus-stop** [comment]
detective supports async/await as of `detective@4.5.0`. i think having an option to configure parsing could make sense though

**Comment by avesus** [comment]
@brandonhorst what does it mean "chokes"? Chokes when you have async/await code in the source tree which `module-deps` parses or it doesn't run with the option `--harmony-async-await` enabled?

I thought that `module-deps` only does comments/quotes syntactic analysis to just discover all `require()` and `import` statements, and that's it. How `async` or `await` statements can break that?..

**Comment by goto-bus-stop** [comment]
module-deps parses files using acorn and walks the syntax tree to find require() calls, so if the parser doesn't support some new syntax it throws.

**Comment by gmurrayhays** [comment]
As a temporary workaround, I went to .../node_modules/falafel/node_modules/acorn/dist/acorn.js and changed the default value of ecmaVersion to 8:

var defaultOptions = {
  // `ecmaVersion` indicates the ECMAScript version to parse. Must
  // be either 3, 5, 6 (2015), 7 (2016), or 8 (2017). This influences support
  // for strict mode, the set of reserved words, and support for
  // new syntax features. The default is 7.
  ecmaVersion: 8,

**Comment by goto-bus-stop** [comment]
@gmurrayhays that looks like a transform you are using does not support recent syntax yet, module-deps itself doesn't use falafel. perhaps you could open up an issue at the relevant transform repo to get it to update!

**Comment by goto-bus-stop** [comment]
https://github.com/browserify/module-deps/commit/2dcc3399ee67ba51ed26d9a0605a8ccdc70c9db7 adds a `detect` option that you can use for this:

```js
var parser = require('module-deps')
var detective = require('detective')

parser({
  detect: function (src) {
    return detective(src, { ecmaVersion: 8 })
  }
})
```

Available in module-deps :package: 6.1.0
